pkg/ebpf/verifier/stats.go

```diff
@@ -206,16 +229,31 @@ func getSourceMap(file string) (map[string]map[int]*SourceLine, error) {
 			continue
 		}
 
-		sourceMap[currSect][insn] = &SourceLine{
+		if functionJustStarted {
+			// llvm-objdump counts instructions since the start of the section, but the verifier
+			// counts from the start of the function. We need to account for that offset
+			insnOffset = insn
+			functionJustStarted = false
+		}
+		insn -= insnOffset
+
+		sourceMap[currFunc][insn] = &SourceLine{
 			LineInfo: currLineInfo,
 			Line:     currLine,
 		}
 	}
 
-	return sourceMap, nil
+	return sourceMap, funcsPerSection, nil
 }
 
-func generateLoadFunction(file string, opts *StatsOptions, stats map[string]*Statistics, complexity map[string]*ComplexityInfo, failedToLoad map[string]struct{}) func(bytecode.AssetReader, manager.Options) error {
+// StatsResult holds the result of the verifier stats process
+type StatsResult struct {
+	Stats           map[string]*Statistics
```

Comment:
seems like you could have a subtype `ObjectFileStats` and have a single `map[string]ObjectFileStats` instead of having each field index on the same thing?

Reply:
`Stats` and `Complexity` are indexed by function name, but we dump all the stats to a single file and the complexity to a file per function (to avoid having a single massive JSON file with all complexity info). And `FuncsPerSection` is indexed by object name and gets dumped to a file per object file (to avoid overwriting that map when we calculate complexity data for a different object).

Reply:
> `Stats` and `Complexity` are indexed by function name

Are they not separated by object file? You can have the same function name is two different object files (and is definitely the case between the non-debug and debug versions of files.

Reply:
Yes, with function name I mean `objectFile/functionName`, it's retaining the same way of indexing as Usama had in the verification stats.

Reply:
OK. I think this could be improved, but can wait for a future PR.